### PR TITLE
Daily log rotation

### PR DESCRIPTION
**Pull request by deepsidhu85**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in configuration to the development, production, and test environments which rotates the log files as they get quite large. Currently this is disabled for development and test environments, but for production the config is set to rotate the files daily. Note that rotate here means creates a new log file and archives the previous day's log.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

An examples of test log files created locally by enabling log rotation for the test environment

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

There are a couple of ways to verify this:

1) Enable the configuration for creating a new development log file in the dev environment, and leave the server running 24+ hours and click through the UI. In the `irida-next-core/log` directory you should now have one `development.log` file and one `development.log.PREVIOUS_DAYS_DATE`

2) 
- Remove existing test.log file from `irida-next-core/logs` if it exists
- Uncomment the `config.logger = ActiveSupport::Logger.new(config.default_log_file, 'daily')` in `environments/test.rb` to allow for log rotation 
- Create a simple test file under `test/logs` with the name log_rotation_test.rb which inherits from `ActiveSupport::TestCase`
- Add a test to the file which does the following:

- Get the first member from the members table and assign to a variable
- Travel 1 day using the `travel` method (https://stackoverflow.com/questions/65834473/how-to-fix-rspec-test-that-depends-on-time-passing-by) and get the first group from the groups table and assign to a variable
- Travel 2 days using the `travel` method above and get the first user from the users table and assign to a variable
- Run the test

If you take a look in the `irida-next-core/logs` directory you should have 3 log files and each of the files should have that corresponding query within

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -106,4 +106,7 @@
 
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
+
+  # Create a new production log file [daily, weekly, monthly,..]
+  config.logger = ActiveSupport::Logger.new(config.default_log_file, 'daily')
```
Comment (ericenns) [inline config/environments/production.rb:111]: **nitpick (blocking):** I wonder if we need to be more like the `RAILS_LOG_TO_STDOUT` block and only enable it when `RAILS_DAILY_LOG_ROTATION` env variable is set. In addition we probably need to make it more like the block above where we also set the log formatter.

**Review by ericenns — CHANGES_REQUESTED** [review]
This looks good, just a minor comment

File: config/environments/production.rb (hunk 1 above)
Comment (deepsidhu85) [reply config/environments/production.rb]: Good call. Updated in e9bf22dce2dd4d421140bc61a576a1d8c7aef510

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**